src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -52,6 +54,16 @@ public StorageOperationException(String message) {
         }
     }
 
+    /**
+     * Signals that some error has occured while trying to makes the storage file read only
+     * while the program is running
+     */
+    public static class StorageReadOnlyException extends Exception {
```

Comment:
Extending from `Exception` may not be the best course of action, as all Exceptions may now be caught under `StorageReadOnlyException` instead. Maybe try to extend the already existing `StorageOperationException`?